api/converter_utils.go

```diff
@@ -401,17 +432,49 @@ func txnRowToTransaction(row idb.TxnRow) (generated.Transaction, error) {
 		localStateDelta = &d
 	}
 
-	txn := generated.Transaction{
+	var logs *[][]byte
+	if len(stxn.ApplyData.EvalDelta.Logs) > 0 {
+		l := make([][]byte, 0, len(stxn.ApplyData.EvalDelta.Logs))
+		for _, v := range stxn.ApplyData.EvalDelta.Logs {
+			l = append(l, []byte(v))
+		}
+		logs = &l
+	}
+
+	var inners *[]generated.TransactionBase
+	if len(stxn.ApplyData.EvalDelta.InnerTxns) > 0 {
+		itxns := make([]generated.TransactionBase, 0, len(stxn.ApplyData.EvalDelta.InnerTxns))
+		for _, t := range stxn.ApplyData.EvalDelta.InnerTxns {
+			extra2 := extra
+			if t.Txn.Type == protocol.ApplicationCallTx {
+				extra2.AssetID = uint64(t.ApplyData.ApplicationID)
+			} else if t.Txn.Type == protocol.AssetConfigTx {
+				extra2.AssetID = uint64(t.ApplyData.ConfigAsset)
+			} else {
+				extra2.AssetID = 0
+			}
```

Comment:
Is it possible to put the intra ID in here too? It isn't strictly required for the REST API, but we'll need it soon when indexing the inner transactions.

Reply:
It is already set to parent's one

Reply:
Yes, but don't we allocate an intra-round offset for each of the inner transactions?

Reply:
I have the same question.

Reply:
Also, can you add a comment that `extra2.AssetCloseAmount` should be 0 here?

Reply:
> > Yes, but don't we allocate an intra-round offset for each of the inner transactions?
> 
> `intra-round offset` is a position in a payset within a block. What value are you expecting for inners?

Sorry, I was mixing up terms here. The "position in a payset within a block" comment is no longer accurate, it was assuming the TxnCounter increments once per transaction in the payset.

The TxnCounter was changed to increment once per Transaction OR inner transaction so that we can create more than one app/asset inside the inner transactions.

The `intra-round` offset should be incremented in the same way.

Reply:
Does it mean all the places that use `TxnRow.Intra` field need to be updated as well?
My thinking was to preserve its meaning in order to find parent within a block if needed. Or are you suggesting tto change `intra-round-offset` only in REST response?

Reply:
We should change how we populate tables `txn` and `txn_participation`, right?

Reply:
then probably this should be bundled with upcoming inner txns indexing

Reply:
No I don't.